openassessment/xblock/grade_mixin.py

```diff
@@ -584,3 +590,79 @@ def _assessment_grade_context(self, assessment):
+        complete = score is not None
+
+        if "staff-assessment" in self.assessment_steps:
+            return "staff"
+
+        # Edge case: staff overrides the grade.
+        # If a score is overriden by staff, it'll always have an
+        # attached annotation type with the `staff_defined` value,
+        # so we look for that in this problem's annotation and
+        # return staff if it's found.
+        grade_annotation_types = [annotation['annotation_type'] for annotation in (score or {}).get("annotations", [])]
+        if complete and "staff_defined" in grade_annotation_types:
+            return "staff"
+
+        # For other cases, we just need to figure out the
+        # priority of each (either peer or self).
+        # Just loop over the values and return the first one
+        # after staff.
+        for _assessment_type in workflow["assessment_score_priority"]:
+            # assessment_step would always have staff in it, so skip it
+            # while checking the priority here.
+            if _assessment_type == "staff":
+                continue
+
+            if "{}-assessment".format(_assessment_type) in self.assessment_steps:
+                return _assessment_type
+
+    def _get_score_explanation(self, workflow):
+        """
+        Return a string which explains how grade is calculated for an ORA assessment
+        (which is complete i.e all assessments have been done) based on assessment_steps.
+        Args:
+            workflow (dict): The serialized Workflow model.
+        Returns:
+            str: Message explainaing how grade is determined.
+        """
+        score = workflow['score']
+        complete = score is not None
+
+        assessment_type = self._get_assessment_type(workflow)
+
+        sentences = {
+            "staff": _("The grade for this problem is determined by your Staff Grade."),
+            "peer": _(
+                "The grade for this problem is determined by the median score of "
+                "your Peer Assessments."
+            ),
+            "self": _("The grade for this problem is determined by your Self Assessment.")
+        }
+        second_sentence = sentences.get(assessment_type, "")
+
+        if complete:
+            first_sentence = _(
+                "You have successfully completed this problem and received a {earned_points}/{total_points}."
+            ).format(earned_points=score["points_earned"], total_points=score["points_possible"])
+        else:
+            first_sentence = ""
+            # Special Case i.e If the submission only have peer assessment
+            if "peer-assessment" in self.assessment_steps and "self-assessment" not in self.assessment_steps and \
+               "staff-assessment" not in self.assessment_steps:
+                first_sentence = _(
+                    "You have not yet received all necessary peer reviews to determine your final grade."
+                )
+                second_sentence = ""
```

Comment:
You are already getting the assessment type above (where the grade is coming from).

With that you can just replace this chunk with:
```suggestion
            # Special Case i.e If the submission only have peer assessment
            if assessment_type == "peer":
                first_sentence = _(
                    "You have not yet received all necessary peer reviews to determine your final grade."
                )
```

Reply:
@ahmedbilal Also, the second sentence should not be empty:
Peer | Peer Step Incomplete | You have not yet received all necessary peer reviews to determine your final grade. | The grade for this problem is determined by the median score of your Peer Assessments.
-- | -- | -- | --

(from the requirements list)

Reply:
**re:** first comment: Yes, it could not work if there is a learning assessment step (use to train student how to correctly assess peer assessments)

Yes, thanks for pointing out. Actually, the upstream ticket is updated many times and then I forget to follow it.